File: backend/app/schemas/system_health.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel

HealthStatus = Literal["ok", "warn", "down"]


class HealthServiceOut(BaseModel):
    name: str
    label: str
    status: HealthStatus
    latency_ms: int | None = None
    detail: str | None = None


class HealthProbeOut(BaseModel):
    name: str
    kind: Literal["page", "api"]
    status: HealthStatus
    status_code: int | None = None
    latency_ms: int | None = None
    detail: str | None = None


class HealthReportOut(BaseModel):
    enabled: bool
    overall: HealthStatus
    services: list[HealthServiceOut]
    pages: list[HealthProbeOut]
    checked_at: datetime

File: frontend/src/api/systemHealth.ts
export type HealthStatus = 'ok' | 'warn' | 'down'

export interface HealthService {
  name: string
  label: string
  status: HealthStatus
  latency_ms?: number
  detail?: string
}

export interface HealthProbe {
  name: string
  kind: 'page' | 'api'
  status: HealthStatus
  status_code: number | null
  latency_ms?: number
  detail?: string
}

export interface HealthReport {
  enabled: boolean
  overall: HealthStatus
  services: HealthService[]
  pages: HealthProbe[]
  checked_at: string
}

export async function fetchSystemHealth(): Promise<HealthReport> {
  const res = await fetch('/api/system-health/', { credentials: 'include' })
  if (!res.ok) throw new Error('Failed to fetch system health')
  return res.json()
}
